dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/ProgramRuleControllerTest.java

```diff
@@ -57,4 +60,20 @@ void testValidateCondition_NoSuchProgram()
         assertWebMessage( "OK", 200, "ERROR", "Expression is not valid",
             POST( "/programRules/condition/description?programId=xyz", "1 != 1" ).content( HttpStatus.OK ) );
     }
+
+    @Test
+    void testDuplicateNameInProgram()
+    {
+        Program program = createProgram( 'A' );
+        manager.save( program );
+        assertStatus( HttpStatus.OK, GET( "/programs/{id}", program.getUid() ) );
+        assertStatus( HttpStatus.OK,
+            POST( "/metadata/", "{'programRules':[{'name':'test', 'program':{ 'id':'" + program.getUid() + "'}}]}" ) );
+
+        JsonImportSummary response = POST( "/metadata/",
+            "{'programRules':[{'name':'test', 'program':{ 'id':'" + program.getUid() + "'}}]}" )
+                .content( HttpStatus.CONFLICT ).get( "response" ).as( JsonImportSummary.class );
+        assertEquals( "The Program Rule name test already exist in Program ProgramA",
+            response.getTypeReports().get( 0 ).getObjectReports().get( 0 ).getErrorReports().get( 0 ).getMessage() );
```

Comment:
FYI: Digging down to a particular report might become unstable should more conflicts be found when new validations are added. There are `find` methods inherited from `JsonObject` where you can essentially grab a conflict by the error code and then check its message. I have done this in some tests.

Reply:
fixed